toptal/picasso
# fix(ScrollMenu): fix scrolling behaviour

[FX-1559]

### Description

Make the scrolling to selected item work with items of different height.

The new implementation doesn't match the old behaviour 100%, but works the same in the majority of cases. If there isn't an issue with it, it can make it work the same but the code wouldn't as small as it is now.

### How to test

- Open Autocomplete storybook
- Go to Scrolling section
- Click on input
- Menu should scroll down to the selected element

- yarn test

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
||![autocomplete-scrolling-to-selected-value-effect-1-snap](https://user-images.githubusercontent.com/77451815/106744860-5a56ca00-6631-11eb-8cb0-1a1efb4882d7.png)|

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run danger` - Danger checks
- `@toptal-bot run lint` - Run linter
- `@toptal-bot run test` - Run jest
- `@toptal-bot run build` - Check build
- `@toptal-bot run test:visual` or `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>


[FX-1559]: https://toptal-core.atlassian.net/browse/FX-1559
